security_monkey/watcher.py

```diff
@@ -295,16 +312,63 @@ def find_modified(self, previous=[], current=[], exception_map={}):
                 self.changed_items.append(eph_change_item)
                 app.logger.debug("%s: changes in item %s/%s/%s" % (self.i_am_singular, eph_change_item.account, eph_change_item.region, eph_change_item.name))
 
-    def find_changes(self, current=[], exception_map={}):
+    def find_changes(self, current=None, exception_map=None):
         """
         Identify changes between the configuration I have and what I had
         last time the watcher ran.
         This ignores any account/region which caused an exception during slurp.
         """
-        prev = self.read_previous_items()
-        self.find_deleted(previous=prev, current=current, exception_map=exception_map)
-        self.find_new(previous=prev, current=current)
-        self.find_modified(previous=prev, current=current, exception_map=exception_map)
+        current = current or []
+        exception_map = exception_map or {}
+
+        # Batching only logic here:
+        if self.batched_size > 0:
+            # Return the items that should be audited:
+            return self.find_changes_batch(current, exception_map)
+
+        else:
+            prev = self.read_previous_items()
+            self.find_deleted(previous=prev, current=current, exception_map=exception_map)
+            self.find_new(previous=prev, current=current)
+            self.find_modified(previous=prev, current=current, exception_map=exception_map)
+
+    def find_changes_batch(self, items, exception_map):
+        # Given the list of items, find new items that don't yet exist:
+        durable_items = []
+
+        # Need to pull out the technology, and the account from the database for
+        # proper queryings...
+        account = datastore.Account.query.filter(Account.name == self.accounts[0]).scalar()
```

Comment:
Replace `scalar()` with `one()` since if technology and account are not found -- that is a problem.